docs/DdnRelocatableShaderElf.md

```diff
@@ -0,0 +1,205 @@
+# Unlinked Shaders And Relocatable Shader Elf
+
+# INTRODUCTION
+
+There is interest in being able to compile SPIR-V shaders without having all of
+the pipeline state information available.  This is advantagous when the same
+shader is used in multiple pipelines because it will only be compiled once.  It
+is also possible to compile the shaders "offline", that is without running the
+full vulkan application. Unlinked and relocatable shaders were designed
```

Comment:
Put this last sentence first.

Basically, restructure this a bit like a design doc: first an overview of what this document is about, some background information, the more detailed objectives and rationale, what it doesn't do or solve, then the technical information.